Code/Framework/AzCore/Tests/ScriptMath.cpp

```diff
@@ -1409,13 +1409,25 @@ namespace UnitTest
         script->Execute("AZTestAssertFloatClose(pl:GetNormal().y,-1)");
         script->Execute("AZTestAssertFloatClose(pl:GetNormal().z,0)");
 
+#if AZ_DEBUG_BUILD
+        AZ_TEST_START_TRACE_SUPPRESSION;
```

Comment:
so this prevents the assert from happening?

Reply:
in unit tests, asserts/errors are intercepted by the test environment (take a look at `TraceBusRedirector` in `Code\Framework\AzCore\AzCore\UnitTest\UnitTest.h`)
This macros allows for the assert/error to be suppressed and then checked that the right amount of them were suppressed. This is basically the tool that its being used to test code that will assert/error